src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -707,25 +705,14 @@ public InsertRowsResponse get() throws Throwable {
               finalResponse.addError(insertError);
             } else {
               boolean changesApplied;
-
-              if (this.enableNesting) {
-                SinkRecord unflattenedRec = this.insertRowsStreamingBuffer.getSinkRecord(originalSinkRecordIdx);
-                changesApplied = SchematizationUtils.evolveSchemaIfNeeded(
+              SinkRecord unflattenedRec = this.insertRowsStreamingBuffer.getSinkRecord(originalSinkRecordIdx);
+              changesApplied = SchematizationUtils.evolveSchemaIfNeeded(
```

Comment:
this makes me 60% nervous but I think its fine simplifying it this way... so instead of using the first row in the buffer to flatten it'll use what we're currently on, which makes sense as we continue to move through the buffer until we get to an apply-able change - this functionality isn't in snowflakes version which is why we need it in all cases

Reply:
can you add a comment explaining for the next person who's struggling to look at this?